Idempotence all grant

azeliashchonak opened the pull request:
##### SUMMARY

In MySQL 8.0 SHOW GRANTS no longer displays ALL PRIVILEGES in its global-privileges output. Instead, SHOW GRANTS explicitly lists each granted global privilege. The issue is that the mysql_user module is using a simple comparison between new and current privileges and if we define ".:ALL,GRANT" it will return changed, because "ALL" != list_of_privileges, so idempotence check will be failed.

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

mysql_user

##### ADDITIONAL INFORMATION

Instead of https://github.com/ansible/ansible/pull/52469

azeliashchonak committed:
init

azeliashchonak committed:
init

azeliashchonak committed:
fix

azeliashchonak committed:
fix

azeliashchonak committed:
fix2

azeliashchonak committed:
fix2

gundalow commented on lib/ansible/modules/database/mysql/mysql_user.py:
```diff
@@ -102,10 +102,8 @@
-
 author:
 - Jonathan Mainguy (@Jmainguy)
-- Benjamin Malynovytch (@bmalynovytch)
```
Why has this line been removed?

azeliashchonak replied:
copy-paste old file) I'll fix it.

gundalow commented on lib/ansible/modules/database/mysql/mysql_user.py:
```diff
@@ -605,9 +583,9 @@ def main():
-            client_cert=dict(type='path', aliases=['ssl_cert']),
-            client_key=dict(type='path', aliases=['ssl_key']),
-            ca_cert=dict(type='path', aliases=['ssl_ca']),
+            ssl_cert=dict(type='path'),
```
Not sure why these aliases have been swapped around?
This change will break people using `client_cert:` in their playbooks

azeliashchonak replied:
fixed

gundalow reviewed (changes requested):
Is it possible your branch wasn't up to date before creating this PR, I see lots of unrelated changed in here

See https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html for guidance on how to rebase your branch

azeliashchonak committed:
update module latest version

azeliashchonak committed:
update module latest version

azeliashchonak committed:
small fixes

azeliashchonak committed:
small fixes

azeliashchonak commented:
/bot_status

azeliashchonak commented:
/ready_for_review

tgadiev commented:
@bmalynovytch can you take a look please?

bmalynovytch commented on lib/ansible/modules/database/mysql/mysql_user.py:
```diff
@@ -115,87 +115,74 @@
     host_all: yes
     state: absent
-
 - name: Create database user with name 'bob' and password '12345' with all database privileges
```
@azeliashchonak 
Many whitespace changes there, probably introduced by automatic formatting.
Could you revert those changes, unless they are necessary / related to the object of the current PR ?

azeliashchonak replied:
Why should we leave empty rows in comments?

bmalynovytch replied:
Oh ! We both misread / misunderstood theses changes: they are not whitespace changes, neither empty rows.
Removing these lines completely changes the structure, removing one level of list (list of lists becomes simple list).
I don't know if this extra level is necessary or not, we need to check how documentation should be provided.

@felixfontein @gundalow Do you have any input about this ?

felixfontein replied:
@azeliashchonak this is not a comment, but extracted and put into the documentation of the module. See here: https://docs.ansible.com/ansible/devel/modules/mysql_user_module.html#examples

In general, examples are better readable if different examples are separated with empty lines. So I would definitely keep them.

azeliashchonak replied:
I've got it, Thanks. Fixed.

bmalynovytch commented on lib/ansible/modules/database/mysql/mysql_user.py:
```diff
@@ -479,7 +471,8 @@ def pick(x):
     for grant in grants:
-        res = re.match("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\5)? ?(.*)""", grant[0])
+        res = re.match(
+            """GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\5)? ?(.*)""", grant[0])
```
There seem to be a regression there 🤔 : `\5)` should be `\6)` as far as I remember

bmalynovytch replied:
I confirm: current version in devel reverted this change, probably when I handled conflicts on my own changes.

bmalynovytch replied:
Fix in review https://github.com/ansible/ansible/pull/55292

azeliashchonak replied:
fixed

bmalynovytch commented on lib/ansible/modules/database/mysql/mysql_user.py:
```diff
@@ -493,14 +486,28 @@ def pick(x):
+
+
+def get_db_version(cursor):
+    cursor.execute("Select @@version")
```
```suggestion
    cursor.execute("SELECT @@version")
```

azeliashchonak replied:
fixed

azeliashchonak committed:
added fix regression #55292

azeliashchonak committed:
added fix regression #55292

azeliashchonak committed:
added fix empty lines

azeliashchonak committed:
added fix empty lines

azeliashchonak committed:
sanity fix

azeliashchonak committed:
sanity fix

azeliashchonak commented:
/ready_for_review

tgadiev commented:
@bmalynovytch can you take a look, pls?

azeliashchonak committed:
Merge branch 'devel' into idempotence-all-grant

Andersson007 committed:
Add unit tests for lib/ansible/module_utils/common/text/converters.py (#56915)

Co-Authored-By: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>

Andersson007 committed:
test_postgres: fix typos

machacekondra committed:
kubevirt: Rewrite merge dict method

Fred-sun committed:
Update azure_rm_dnsrecordset relate document (#56895)

* Update azure_rm_dnsrecordset related documentation

Fred-sun committed:
Update azure_rm_dnszone related document (#56984)

*  Update azure_rm_dnszone related documentation

Fred-sun committed:
Update azure_rm_functionapp related document (#56985)

* Update azure_rm_functionapp related documentation

Fred-sun committed:
Update azure_rm_loadbalancer relate module document (#57154)

richlv committed:
suggest to use sequential IDs (#57142)

##### SUMMARY
Suggest to use sequential IDs - otherwise action is considered to be changed always.


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

Fred-sun committed:
Update azure_rm_hdinsightcluster related document (#57037)

* Update azure_rm_hdinsightcluster related documentation

webknjaz committed:
Fix ansible_async_dir in the migration doc

Fred-sun committed:
Update azure_rm_mariadbconfiguration related document (#57207)

evrardjp committed:
Removing myself from openstack team (#57212)

I guess you might have seen that I do not have much bandwidth to
maintain those modules. I want to step up for other modules, so
I will have to choose. The OpenStack modules are in good hands
anyway with the rest of the team_openstack :)

Sorry for the inconvenience.

Commit:
Issue 56430 (#57147)

* Adding integration test for 127.0.0.1/32 and ::1/128.

* Making sure file is not corrupted when render fails

* Fixes #56430

* Adding changelog for MR 57147/Issue 56430

rohitthakur2590 committed:
vyos interfacefix and  test cases (#57169)

Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

webknjaz committed:
Fix module_utils.common.test_collections testsuite

PR #57181

azeliashchonak committed:
Merge branch 'idempotence-all-grant' of https://github.com/lean-delivery/ansible into idempotence-all-grant

Andersson007 commented:
@azeliashchonak , please solve conflicts or close this PR

needs_info

bmalynovytch commented:
Branch primarily needs a rebase (which will require to solve conflicts)
